lib/pickle/session/parser.rb

```diff
@@ -24,7 +24,7 @@ def match_field
 
       def parse_hash(hash)
         hash.inject({}) do |parsed, (key, val)|
-          if session && val =~ /^#{capture_model}$/
+          if session && !val.is_a?(Array) && val =~ /^#{capture_model}$/
```

Comment:
@mathieujobin I'm seeing this error also for `Hash`.  Could this be changed to only perform the regex check if `val.is_a?(String)`?

Reply:
@ejennings-mdsol yes it can, I simply preferred to make as little behavioral changes as possible.
If you think it's safe to do it on `String` only. we can update it.
I am also tempted to add the exclusion on `Hash` as well, so that if `val` is something else, we would know

Reply:
You are correct... I made this change, will be released as v0.7.2